src/common/attachments/uploadAttachmentsDialog.component.test.tsx

```diff
@@ -0,0 +1,229 @@
+    createView();
+
+    await user.click(screen.getByLabelText('Close Modal'));
+
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it('posts attachment metadata successfully', async () => {
+    // Render the component
+
+    createView();
+
+    const file1 = new File(['test'], 'test1.txt', {
+      type: 'text/plain',
+    });
+
+    // Find the Uppy Dashboard's drop zone (it usually has a label like "Drop files here" or "Browse files")
+    const dropZone = screen.getByText('Files cannot be larger than', {
+      exact: false,
+    });
+
+    // Create a drag-and-drop event for the file
+    Object.defineProperty(dropZone, 'files', {
+      value: [file1],
+    });
+
+    // Fire the drop event
+
+    fireEvent.drop(dropZone, {
+      dataTransfer: {
+        files: [file1],
+      },
+    });
+
+    // Wait for the UI to update with the added file
+    await waitFor(() => {
+      expect(screen.getByText('test1.txt')).toBeInTheDocument();
+    });
+
+    await user.click(await screen.findByText('Upload 1 file'));
+
+    // Assert axios post was called
+    expect(axiosPostSpy).toHaveBeenCalledWith('/attachments', {
+      entity_id: '1',
+      file_name: 'test1.txt',
+    });
+
+    expect(xhrPostSpy).toHaveBeenCalledWith(
+      'POST',
+      'http://localhost:3000/object-storage',
+      true
+    );
+
+    expect(await screen.findByText('Complete')).toBeInTheDocument();
+  });
+
+  it('errors when presigned url fails', async () => {
+    server.use(
+      http.post('/object-storage', async () => {
+        return HttpResponse.error();
+      })
+    );
+
+    createView();
+
+    const file1 = new File(['test'], 'uploadError.txt', {
+      type: 'text/plain',
+    });
+
+    // Find the Uppy Dashboard's drop zone (it usually has a label like "Drop files here" or "Browse files")
+    const dropZone = screen.getByText('Files cannot be larger than', {
+      exact: false,
+    });
+
+    // Create a drag-and-drop event for the file
+    Object.defineProperty(dropZone, 'files', {
+      value: [file1],
+    });
+
+    // Fire the drop event
+
+    fireEvent.drop(dropZone, {
+      dataTransfer: {
+        files: [file1],
+      },
+    });
+
+    // Wait for the UI to update with the added file
+    await waitFor(() => {
+      expect(screen.getByText('uploadError.txt')).toBeInTheDocument();
+    });
+
+    await user.click(await screen.findByText('Upload 1 file'));
+
+    // Assert axios post was called
+    expect(axiosPostSpy).toHaveBeenCalledWith('/attachments', {
+      entity_id: '1',
+      file_name: 'uploadError.txt',
+    });
+
+    expect(xhrPostSpy).toHaveBeenCalledWith(
+      'POST',
+      'http://localhost:3000/object-storage',
+      true
+    );
+
+    expect(await screen.findByText('Upload failed')).toBeInTheDocument();
+
+    expect(
+      await screen.findByLabelText('Show error details')
+    ).toBeInTheDocument();
+  });
+
+  it('errors when file is removed mid upload', async () => {
```

Comment:
Sorry one last thing - is this really an error?
```suggestion
  it('can remove a file mid upload', async () => {
```
I think this was repeated in the e2e test as well?

Reply:
'should send a DELETE request for the attachment document if a file is removed during upload'